File: app/src/main/java/be/hogent/yasminedewinne/carwashapp/models/domain/Afspraak.kt
package be.hogent.yasminedewinne.carwashapp.models.domain

import androidx.room.*
import be.hogent.yasminedewinne.carwashapp.util.converters.DateConverter
import be.hogent.yasminedewinne.carwashapp.util.converters.TimeConverter
import java.time.LocalDate
import java.time.LocalTime

@Entity(
    tableName = "afspraken",
    indices = [
        (Index(value = ["id"], unique = true))
    ]
)
@TypeConverters(value = [DateConverter::class, TimeConverter::class])
data class Afspraak (

    @PrimaryKey(autoGenerate = false)
    var id: Int = 0,

    var gebruikerId: Int,

    var carwashId: Int,
    var carwashAdres: String,
    var carwashMerk: String,
    var carwashAuto: String,
    var carwashTarief: Int,
    var carwashDatum: LocalDate,
    var carwashBeginTijd: LocalTime,
    var carwashEindTijd: LocalTime
)

File: API/CarwashAPI/Models/Domain/Afspraak.cs
using System;
namespace CarwashAPI.Models.Domain
{
    public class Afspraak
    {
        public int Id { get; set; }

        public int GebruikerId { get; set; }
        public User Gebruiker { get; set; }
        public int CarwashId { get; set; }
        public Carwash Carwash { get; set; }

        public Afspraak(){}
        public Afspraak(int userId, int carwashId)
        {
            this.GebruikerId = userId;
            this.CarwashId = carwashId;
        }
    }
}
